test/test_transformer.cc

```diff
@@ -518,3 +518,211 @@ TEST(Transformer, Codegen) {
   ASSERT_TRUE(loop4pos > loop3pos);
   ASSERT_TRUE(stmtpos > loop4pos);
 }
+
+static isl::schedule_node tile(isl::schedule_node node,
+			       std::vector<int> tileSizes,
+  			       int defaultSize = 32) {
+  isl::space space = isl::manage(isl_schedule_node_band_get_space(node.get()));
+  unsigned dims = space.dim(isl::dim::set);
+  isl::multi_val sizes = isl::multi_val::zero(space);
+  for(unsigned i = 0; i < dims; ++i) {
+    int tileSize = i < tileSizes.size() ? tileSizes[i] : defaultSize;
+    sizes = sizes.set_val(i, isl::val(node.get_ctx(), tileSize));
+  }
+  node = 
+    isl::manage(isl_schedule_node_band_tile(node.release(), sizes.release()));
+  return node;
+}
+
+static isl::multi_union_pw_aff permuteDims(isl::multi_union_pw_aff ps, int firstDim,
+					   int secondDim) {
+  auto scheduleFirstDim = ps.get_union_pw_aff(firstDim);
+  auto scheduleSecondDim = ps.get_union_pw_aff(secondDim);
+  ps = ps.set_union_pw_aff(secondDim, scheduleFirstDim);
+  ps = ps.set_union_pw_aff(firstDim, scheduleSecondDim);
+  return ps;
+}
+
+TEST(Transformer, matmul) {
```

Comment:
Please capitalize the name of the test, it may also be helpful to make it more self-descriptive like `MatchMatmul`